packages/lib-ethers/utils/deploy.ts

```diff
@@ -308,97 +280,248 @@ const connectContracts = async (
-  _priceFeedIsTestnet = true,
+  contractsVersion: string,
+  _useRealPriceFeed: boolean,
   _isDev = true,
   overrides?: Overrides
-): Promise<_LiquityDeploymentJSON> => {
+): Promise<_LiquityDeploymentJSON[]> => {
   if (!deployer.provider) {
     throw new Error("Signer must have a provider.");
   }
 
   log("Deploying contracts...");
   log();
 
-  const deployment: _LiquityDeploymentJSON = {
+  const deployments: _LiquityDeploymentJSON[] = []
+
+  let firstDeployment: _LiquityDeploymentJSON = {
     chainId: await deployer.getChainId(),
-    version: "unknown",
+    version: contractsVersion,
+    collateralSymbol: firstCollateralSymbol,
     deploymentDate: new Date().getTime(),
-    _priceFeedIsTestnet,
+    _useRealPriceFeed,
     _isDev,
 
-    ...(await deployContracts(deployer, oracleAddresses, collateralSymbol, collateralAddress, getContractFactory, delay, stablecoinAddress, _priceFeedIsTestnet, overrides).then(
+    ...(await deployContracts(
+      deployer, 
+      oracleAddresses, 
+      firstCollateralSymbol, 
+      firstCollateralAddress, 
+      getContractFactory, 
+      delay, 
+      _useRealPriceFeed, 
+      overrides
+    ).then(
       async ([addresses, startBlock]) => ({
         startBlock,
 
         addresses: {
-          ...addresses
+          ...addresses,
+          thusdToken: stablecoinAddress ? stablecoinAddress : "",
+          bamm: stablecoinAddress 
+          ? await deployContract(
+              deployer, 
+              getContractFactory, 
+              "BAMM",
+              addresses.chainlink,
+              addresses.stabilityPool,
+              stablecoinAddress,
+              addresses.erc20,
+              400,
+              process.env.BAMM_FEE_POOL || contractOwners["feePool"],
+              process.env.BAMM_OWNER || contractOwners["bammOwner"],
+              { ...overrides }
+            )
+          : ""
         }
       })
     ))
   };
 
-  const contracts = _connectToContracts(deployer, deployment);
+  if (!stablecoinAddress) {
+    let thusdTokenAddress = ""
+
+    const secondDeployment: _LiquityDeploymentJSON = {
+      chainId: await deployer.getChainId(),
+      version: contractsVersion,
+      collateralSymbol: secondCollateralSymbol,
+      deploymentDate: new Date().getTime(),
+      _useRealPriceFeed,
+      _isDev,
+  
+      ...(await deployContracts(
+        deployer, 
+        oracleAddresses, 
+        secondCollateralSymbol, 
+        secondCollateralAddress, 
+        getContractFactory, 
+        delay, 
+        _useRealPriceFeed, 
+        overrides
+      ).then(
+        async ([addresses, startBlock]) => {
+          
+          const thusdToken = await deployContract(
+            deployer,
+            getContractFactory,
+            "THUSDToken",
+            firstDeployment.addresses.troveManager,
+            firstDeployment.addresses.stabilityPool,
+            firstDeployment.addresses.borrowerOperations,
+            addresses.troveManager,
```

Comment:
naming a bit confusing, why some called `firstDeployment.addresses` and other just `addresses`?

Reply:
Good point, although it is technically correct. it lacks clarity on what concerns the different deployment sets identification. Resolved in [feat: add pcv initialization in deployment scripts](https://github.com/Threshold-USD/dev/pull/102/commits/42f0ac098fdf83eef4a659e32dd25867d659e704)